google/cloud/aiplatform/experimental/vertex_model/serializers/model.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pathlib
+import tempfile
+import torch
+
+from google.cloud import storage
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+
+
+def _serialize_local_model(artifact_uri: str, obj: torch.nn.Module, model_type: str):
+    """Serializes torch.nn.Module object to GCS.
+
+    Args:
+        artifact_uri (str): the GCS bucket where the serialized object will reside.
+        obj (torch.nn.Module): the model to serialize.
+        dataset_type (str): the model name and usage
+
+    Returns:
+        The GCS path pointing to the serialized objet.
+    """
+
+    compiled_custom_model = torch.jit.script(obj)
+
+    if artifact_uri[0:6] != "gs://":
+        path_to_model = artifact_uri + "/my_" + model_type + "_model.pth"
+        torch.jit.save(compiled_custom_model, path_to_model)
+        return path_to_model
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
It seems like this block could also call `serialize_to_tmp_and_copy_to_gcs`. It might be useful to move that method into a generic utilities module and call it here as well.